presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoStatement.java

```diff
@@ -170,10 +166,66 @@ public void setCursorName(String name)
     public boolean execute(String sql)
             throws SQLException
     {
+        clearCurrentResults();
+
         checkOpen();
-        // Only support returning a single result set
-        currentResult.set(executeQuery(sql));
-        return true;
+        requireNonNull(sql, "null sql statement");
+
+        StatementClient client = null;
```

Comment:
done for StatementClient. Cant use for resultSet as it is closed only if currentResult was not set.